pkg/handler/middleware.go

```diff
@@ -0,0 +1,45 @@
+package handler
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/sirupsen/logrus"
+)
+
+// Middleware для проверки аутентификации
+func AuthMiddleware(c *gin.Context) {
+	tokenString, err := c.Cookie("token")
+	if err != nil {
+		c.JSON(http.StatusUnauthorized, gin.H{"error": "Неавторизован"})
+		c.Abort()
+		return
+	}
+
+	logrus.Infoln("token: ", tokenString)
+
+	claims := &Claims{}
+	token, err := jwt.ParseWithClaims(tokenString, claims, func(token *jwt.Token) (interface{}, error) {
+		return []byte(signingKey), nil
+	})
+
+	if err != nil {
+		if err == jwt.ErrSignatureInvalid {

```

Comment:
Для проверки ошибок нужно использовать специализированную функцию
```suggestion
		if errors.Is(err, jwt.ErrSignatureInvalid) {
```